extensions/trace_propagators/src/main/java/io/opentelemetry/extensions/trace/propagation/B3Propagator.java

```diff
@@ -31,44 +58,54 @@
   private static final List<String> FIELDS =
       Collections.unmodifiableList(Arrays.asList(TRACE_ID_HEADER, SPAN_ID_HEADER, SAMPLED_HEADER));
 
+  private static final B3Propagator INSTANCE = B3Propagator.builder().build();
+
+  private final B3PropagatorExtractor singleHeaderExtractor =
+      new B3PropagatorExtractorSingleHeader();
+  private final B3PropagatorExtractor multipleHeadersExtractor =
+      new B3PropagatorExtractorMultipleHeaders();
   private final B3PropagatorInjector b3PropagatorInjector;
-  private final B3PropagatorExtractor b3PropagatorExtractor;
 
-  private static final B3Propagator SINGLE_HEADER =
-      new B3Propagator(
-          new B3PropagatorInjectorSingleHeader(), new B3PropagatorExtractorSingleHeader());
-  private static final B3Propagator MULTI_HEADER =
-      new B3Propagator(
-          new B3PropagatorInjectorMultipleHeaders(), new B3PropagatorExtractorMultipleHeaders());
+  private B3Propagator(B3PropagatorInjector b3PropagatorInjector) {
+    this.b3PropagatorInjector = b3PropagatorInjector;
+  }
 
   /**
-   * Returns an instance of {@link B3Propagator} with Single Header Implementation of B3 propagation
-   * protocol. See <a
-   * href=https://github.com/openzipkin/b3-propagation#single-header>openzipkin/b3-propagation#single-header</a>.
-   *
-   * @return Returns an instance of {@link B3Propagator} with Single Header implementation of B3
-   *     propagation protocol.
+   * Returns a new {@link B3Propagator.Builder} instance for configuring injection option for {@link
+   * B3Propagator}.
    */
-  public static B3Propagator getSingleHeaderPropagator() {
-    return SINGLE_HEADER;
+  public static B3Propagator.Builder builder() {
+    return new Builder();
   }
 
   /**
-   * Returns an instance of {@link B3Propagator} with Multiple Header Implementation of B3
-   * propagation protocol. See <a
-   * href=https://github.com/openzipkin/b3-propagation#multiple-headers>openzipkin/b3-propagation#multiple-headers</a>.
-   *
-   * @return Returns an instance of {@link B3Propagator} with Multiple Header implementation of B3
-   *     propagation protocol.
+   * Enables the creation of an {@link B3Propagator} instance with the ability to switch injector
+   * from single header (default) to multiple headers.
    */
-  public static B3Propagator getMultipleHeaderPropagator() {
-    return MULTI_HEADER;
+  public static class Builder {
+    private boolean injectSingleHeader;
+
+    private Builder() {
+      injectSingleHeader = true;
+    }
+
+    public Builder injectFromMultipleHeaders() {
```

Comment:
Inject happens to headers, not from. But how about just `injectMultipleHeaders`?

Reply:
sounds good, i'll tidy it up